### PR TITLE
Add kubernetes config docs #1072

### LINKED ISSUE #1072: Kubernetes missing from configuration guide
@MatthewBM reports that the configuration section is missing docs for Kubernetes.

### PR DESCRIPTION
**Pull request by ZhuozhaoLi**
fix #1072

### CODE REVIEW

File: parsl/configs/kubernetes_htex.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            max_workers=1,
+            worker_logdir_root='runinfo',
+            address=address_by_route(),  # Address for the pod worker to connect back
+            provider=KubernetesProvider(
+                namespace="default",
+                image='CONTAINER_LOCATION',  # Specify where to download the image
+                nodes_per_block=1,
+                init_blocks=1,  
+                max_blocks=10,  # Maximum number of pods to scale up
+                worker_init="""pip install git+https://github.com/Parsl/parsl""", # install Parsl when the pod starts
```
Comment (benclifford) [inline parsl/configs/kubernetes_htex.py]: The azure and AWS providers install the latest release version of parsl (in template.py). For consistency, maybe this should also install that version, not the github master version.

File: docs/userguide/configuring.rst [hunk 2]
```diff
@@ -304,6 +304,17 @@ re-used in subsequent runs. The configuration uses the `AWSProvider` to connect
 
 .. literalinclude:: ../../parsl/configs/ec2_htex_single_node.py
 
+Kubernetes Clusters
+-------------------
+
+.. image:: https://d1.awsstatic.com/PAC/kuberneteslogo.eabc6359f48c8e30b7a138c18177f3fd39338e05.png
+
+Kubernetes is an open-source system for container management, such as automating deployment and scaling of containers.
+The snippet below shows an example configuration for deploying non-deployment pods as workers on a Kubernetes cluster.
```
Comment (benclifford) [inline docs/userguide/configuring.rst]: I don't think the phrase "non-deployment" is needed here. Not-using-deployments is something from history, not from the current version of parsl.

File: parsl/configs/kubernetes_htex.py (hunk 1 above)
Comment (ZhuozhaoLi) [reply parsl/configs/kubernetes_htex.py]: Change to install from pypi now

File: docs/userguide/configuring.rst (hunk 2 above)
Comment (ZhuozhaoLi) [reply docs/userguide/configuring.rst]: Removed `non-deployment`

**Comment by benclifford** [comment]
This is failing on flake8 still:

```
$ flake8 parsl/
parsl/configs/kubernetes_htex.py:8:14: E251 unexpected spaces around keyword / parameter equals
parsl/configs/kubernetes_htex.py:8:16: E251 unexpected spaces around keyword / parameter equals
parsl/configs/kubernetes_htex.py:19:31: W291 trailing whitespace
parsl/configs/kubernetes_htex.py:28:1: W391 blank line at end of file
The command "flake8 parsl/" exited with 1.
```

**Pull request merged**